OneCommunityGlobal/HighestGoodNetworkApp
# Carlos new pie chart hour visualization by team member  project reports

(PRIORITY HIGH) Jae: Add “Hours by Team Member” Visualization to the Projects Reports page (WIP Carlos)

Related PRS (if any):
Previous PR2310
This current is related to Backend PR [#1015](https://github.com/OneCommunityGlobal/HGNRest/pull/1015)

Description
The graph currently displays weekly committed hours by project member. However, it does not show the actual hours worked by member on each project.

Main changes explained:
in getTimeEntriesForSpecifiedProjectadd, add firstName lastName and isActive properties in 'TimeEntry' Schema from 'userProfile' Schema with will be received in the fronted.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as Owner, manager or admin user
5. Go to dashboard => report => type the project => PieChart => pick the charts.
6. check and test the pie charts as required

Requested job:

![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55204862/5fe12ace-d7d4-488f-b081-79be177aeeed)


video before the change:

https://www.loom.com/share/7be14c4aec464c5fbb9669a11c4d63b5?sid=586d7ba2-8f94-4a4a-99e6-6725b31d4236

video after the change:

https://www.loom.com/share/478133a664ff48cd832d86abdef3edd4?sid=3222bf49-79f9-4fd8-b1d8-abe4bb0687b2

video after the change (toggle type)

https://www.loom.com/share/b4c44576df94464996eb0ef2321c4528?sid=7a1ebf85-e192-451a-b39b-1de04a6cb891

Note:
You must add a project for at least one user ( set as a poject member)
You must log hours for this member(s) in this project.
Go to dashboard => report => type the project => PieChart => pick the charts
4.See videos above for more info
